dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1072,6 +1072,18 @@ public List<Workflow> getAllWorkflowByPath(@ApiParam(hidden = true) @Parameter(h
         return workflows;
     }
 
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/path/{repository}/published")
+    @Operation(operationId = "getAllPublishedWorkflowByPath", summary = "Get a list of published workflows by path.", description = "Do not include workflow name.",
+            security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    public List<Workflow> getAllPublishedWorkflowByPath(@Parameter(description = "repository path", required = true) @PathParam("repository") String path) {
+        List<Workflow> workflows = Optional.ofNullable(workflowDAO.findAllByPath(path, true)).orElse(List.of());
+        workflows.forEach(this::checkNotNullEntry);
```

Comment:
Nitpick, do either of the `findAllByPath` methods actually return a list  with a null in it (as opposed to a null or an empty list) ? 
That kinda seems like a bug that should be addressed / ticket created (and this  workaround can be tagged with a TODO).

Reply:
Tested it and it doesn't return a list with a null, just null if there are no results.

To clarify, the ticket should address the unneeded `checkNotNullEntry` calls?

Reply:
Hmmm, or could remove here now. 
I just wasn't sure if you encountered this specifically, so put this guard in. 
Or was it copied from a different endpoint where it is also not necessary?

Reply:
It was copied from [another endpoint](https://github.com/dockstore/dockstore/blob/0c1fa475cf7980e8244750d2c6ce9b7169927f7b/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L1067) where it's also not necessary. Should I create a different ticket to address that?

Reply:
Ah ok, yeah ticket would be fine

Reply:
@kathy-t FYI for link

Reply:
https://ucsc-cgl.atlassian.net/browse/SEAB-6177